linera-chain/src/block.rs

```diff
@@ -17,11 +17,31 @@ use thiserror::Error;
 use crate::{data_types::ExecutedBlock, ChainError};
 
 /// Wrapper around an `ExecutedBlock` that has been validated.
-#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[derive(Debug, PartialEq, Eq, Clone)]
 pub struct ValidatedBlock {
     executed_block: Hashed<ExecutedBlock>,
 }
 
+impl Serialize for ValidatedBlock {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        self.executed_block.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for ValidatedBlock {
+    fn deserialize<D>(deserializer: D) -> Result<ValidatedBlock, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        Ok(ValidatedBlock {
+            executed_block: Hashed::deserialize(deserializer)?,
+        })
+    }
+}
+
 impl ValidatedBlock {
```

Comment:
Do we need the type `ValidatedBlock` and `ConfirmedBlock` and all this boilerplate in the first place?

Reply:
I've thought about that (i.e. removal of `ValidatedBlock` and `ConfirmedBlock`) but decided not to do that (yet) b/c:
- they still seem useful when distinguishing between different type of blocks we're dealing with in various steps of the protocol flow
- we'd have to decide what to do with the `Timeout` type which is a corresponding type for the `TimeoutCertificate`.
- didn't want to do that now as it seemed not a priority.